includes/core/DataTransferObjects/class-data-transfer-object.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Base DTO.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\DataTransferObjects;
+
+use ReflectionClass;
+use ReflectionException;
+
+/**
+ * Abstract DTO class.
+ */
+abstract class Data_Transfer_Object {
```

Comment:
As I mentioned in another comment, I'd suggest focusing on solidifying how we interact with the server and leaving the introduction of DTO to another issue. 

We can just move these methods to the `Response` class which so far in this PR is the only class that extends `Data_Transfer_Object` and focus on the API for server interactions.

If needs be, we can always refactor these methods out into DTO base class later.

Reply:
Make sense. I will remove all DTO abstractions into parent Request/Response class. If needed, we could easily extract logic to separate class if we need it in the future.

Reply:
Fixed.